packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/BacklinksTests.kt

```diff
@@ -278,24 +336,29 @@ class BacklinksTests {
                 parent.child = child
                 parent.childList.add(child)
                 parent.childSet.add(child)
+                parent.childDictionary["A"] = child
             }
 
             child.assertParents(parents.size)
         }
 
-        realm.query<Child>().first().find()!!.let { child ->
-            assertEquals(2, child.parents.query("id > 2").find().size)
-            assertEquals(2, child.parentsByList.query("id > 2").find().size)
-            assertEquals(2, child.parentsBySet.query("id > 2").find().size)
-        }
+        realm.query<Child>()
+            .first()
+            .find { child ->
+                assertNotNull(child)
+                assertEquals(2, child.parents.query("id > 2").find().size)
+                assertEquals(2, child.parentsByList.query("id > 2").find().size)
+                assertEquals(2, child.parentsBySet.query("id > 2").find().size)
+                assertEquals(2, child.parentsByDictionary.query("id > 2").find().size)
+            }
     }
 
     @Test
     fun deleteSourceObject() {
         val child = realm.writeBlocking {
             val child = this.copyToRealm(Child())
 
-            val parents = Array(5) {
+            val parents = (1..5).map {
```

Comment:
The problem with `repeat` is that it doesn't do any mapping and we would have to add the output manually for each iteration. Otherwise I agree, it would look better.